### PR TITLE
Refactor lints in methods module

### PR DESCRIPTION
**Pull request by TaKO8Ki**
This PR refactors methods lints other than the lints I refactored in https://github.com/rust-lang/rust-clippy/pull/6826 and moves some functions to methods/utils.rs.
Basically, I follow the instruction described in #6680.

**For ease of review, I refactored step by step, keeping each commit small.**

closes https://github.com/rust-lang/rust-clippy/issues/6886
cc: @phansch, @flip1995, @Y-Nak 

changelog: Move lints in methods module to their own modules and some function to methods/utils.rs.

### CODE REVIEW

**Comment by TaKO8Ki** [comment]
@phansch, @flip1995, @Y-Nak
I'm ready for your review!

File: clippy_lints/src/methods/into_iter_on_ref.rs [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+    expr: &hir::Expr<'_>,
+    method_span: Span,
+    method_name: Symbol,
+    args: &[hir::Expr<'_>],
+) {
+    let self_ty = cx.typeck_results().expr_ty_adjusted(&args[0]);
+    if_chain! {
+        if let ty::Ref(..) = self_ty.kind();
+        if method_name == sym::into_iter;
+        if match_trait_method(cx, expr, &paths::INTO_ITERATOR);
```
Comment (Y-Nak) [inline clippy_lints/src/methods/into_iter_on_ref.rs:25]: We are removing the use of `paths`.
```suggestion
        if is_trait_method(cx, expr, sym::IntoIterator);
```

**Comment by flip1995** [comment]
Sorry for the merge conflicts. Resolving those should be quite easy though, since  #6918 only changed some imports.

File: clippy_lints/src/methods/get_unwrap.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use crate::methods::derefs_to_slice;
+use crate::methods::utils::derefs_to_slice;
```
[the remark this answers is not in the document]
Comment (TaKO8Ki) [reply clippy_lints/src/methods/get_unwrap.rs]: I fixed that!

[the remark this answers is not in the document]
Comment (Y-Nak) [reply clippy_lints/src/methods/get_unwrap.rs]: It seems `use crate::method::utils` still exists in this  PR.

[the remark this answers is not in the document]
Comment (TaKO8Ki) [reply clippy_lints/src/methods/get_unwrap.rs]: Yeah. I've not pushed yet.

[the remark this answers is not in the document]
Comment (TaKO8Ki) [reply clippy_lints/src/methods/get_unwrap.rs]: I'm sorry I didn't explain it enough. I just told you that I committed some changes in order to accept your suggestion.

File: clippy_lints/src/methods/chars_cmp.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_lint::LateContext;
+use rustc_lint::Lint;
+use rustc_middle::ty;
+use rustc_span::sym;
+
+/// Wrapper fn for `CHARS_NEXT_CMP` and `CHARS_LAST_CMP` lints.
+pub(super) fn check(
```
[the remark this answers is not in the document]
Comment (TaKO8Ki) [reply clippy_lints/src/methods/chars_cmp.rs:13]: I agree with you. I'm going to merge `chars_last_cmp` and `chars_next_cmp` and merge `chars_next_cmp_with_unwrap` and `chars_last_cmp_with_unwrap`.

**Comment by phansch** [comment]
@bors r+

Thanks for your patience!

File: clippy_lints/src/methods/chars_cmp.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (TaKO8Ki) [reply clippy_lints/src/methods/chars_cmp.rs:13]: I'm going to do it in a new PR.

**Pull request merged**